src/Lykke.AzureStorage/Tables/Entity/Metamodel/EntityMetamodelImpl.cs

```diff
@@ -23,12 +26,91 @@ public IStorageValueSerializer TryGetSerializer(PropertyInfo property)
                 // Serializer defined on the property, overrides that one, defined on the type, which property has
 
                 return _provider.TryGetPropertySerializer(property) ??
-                       _provider.TryGetTypeSerializer(property.PropertyType);
+                       _provider.TryGetTypeSerializer(property.PropertyType) ??
+                       TryGetCollectionSerializer(property.PropertyType) ??
+                       TryGetNullableSerializer(property.PropertyType);
+
             }
             catch (Exception ex)
             {
                 throw new InvalidOperationException($"Failed to get storage value serializer for property {property.DeclaringType}.{property.Name}, in context of type {property.ReflectedType}", ex);
             }
         }
+
+        public ValueTypeMergingStrategy? TryGetValueTypeMergingStrategy(Type type)
+        {
+            try
+            {
+                return _provider.TryGetValueTypeMergingStrategy(type);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException($"Failed to get value type merging strategy factory for entity type {type}", ex);
+            }
+        }
+
+        private IStorageValueSerializer TryGetCollectionSerializer(Type collectionType)
+        {
+            var enumerbleGenericArguments = GetGenericArgumentsOfAssignableType(collectionType, typeof(IEnumerable<>));
+            if (enumerbleGenericArguments != null)
+            {
+                var elementType = enumerbleGenericArguments.First();
+
+                return _provider.TryGetTypeSerializer(elementType);
+            }
+
+            if (collectionType.IsArray)
+            {
+                var elementType = collectionType.GetElementType();
+
+                return _provider.TryGetTypeSerializer(elementType);
+            }
+
+            return null;
+        }
+
+        private IStorageValueSerializer TryGetNullableSerializer(Type nullableType)
+        {
+            var nullableGenericArguments = GetGenericArgumentsOfAssignableType(nullableType, typeof(Nullable<>));
+            if (nullableGenericArguments != null)
+            {
+                var elementType = nullableGenericArguments.First();
+
+                return _provider.TryGetTypeSerializer(elementType);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Returns generic arguments of type <paramref name="genericType"/>, 
+        /// if <paramref name="givenType"/> is assignable to the <paramref name="genericType"/>,
+        /// or null
+        /// </summary>
+        private static Type[] GetGenericArgumentsOfAssignableType(Type givenType, Type genericType)
```

Comment:
* ReSharper suggests to convert recursive call into a loop.
* Why the return value is not IReadOnlyList?

Reply:
Fixed